playbooks/roles/streisand-gateway/tasks/main.yml

```diff
@@ -101,63 +101,65 @@
+        dest: "{{ item.dest }}"
+      with_items:
+        - { src: "index.md.j2",        dest: "{{ streisand_gateway_markdown_index }}" }
+        - { src: "instructions.md.j2", dest: "{{ streisand_gateway_markdown_instructions }}" }
+        - { src: "firewall.md.j2",     dest: "{{ streisand_gateway_markdown_firewall_information }}" }
+
+    - name: Convert the Markdown Gateway index, instructions, and firewall information into HTML and surround them with the header and footer template
+      shell: markdown {{ item.md }} | cat {{ streisand_header_template }} - {{ streisand_footer_template }} > {{ item.html }}
+      with_items:
+        - { md: "{{ streisand_gateway_markdown_index }}",                html: "{{ streisand_gateway_html_index }}" }
+        - { md: "{{ streisand_gateway_markdown_instructions }}",         html: "{{ streisand_gateway_html_instructions }}" }
+        - { md: "{{ streisand_gateway_markdown_firewall_information }}", html: "{{ streisand_gateway_html_firewall_information }}" }
+
+    - name: Fetch the Gateway HTML instructions and firewall information
+      fetch:
+        dest: ../{{ streisand_local_directory }}/
+        src: "{{ item }}"
+        flat: yes
+      with_items:
+        - "{{ streisand_gateway_html_instructions }}"
+        - "{{ streisand_gateway_html_firewall_information }}"
+
+    - name: Clean up the temporary Gateway instruction and firewall information files once they have been retrieved
+      file:
+        path: "{{ item }}"
+        state: absent
+      with_items:
+        - "{{ streisand_gateway_html_instructions }}"
+        - "{{ streisand_gateway_markdown_instructions }}"
+        - "{{ streisand_gateway_html_firewall_information }}"
+        - "{{ streisand_gateway_markdown_firewall_information }}"
+
+    - name: Ensure that all of the files in the Gateway have the proper permissions
+      file:
+        path: "{{ streisand_gateway_location }}"
+        owner: www-data
+        group: www-data
+        mode: 0750
+        state: directory
+        recurse: yes
+
+    - meta: flush_handlers
+
+    - name: Success!
+      pause:
+        prompt: "Server setup is complete. The `{{ streisand_server_name }}.html` instructions file in the generated-docs folder is ready to give to friends, family members, and fellow activists. Press Enter to continue."
+
+    - name: Attempt to open the instructions on Linux (if applicable). Errors in this task are ignored because the `xdg-open` command is not always available.
+      local_action: command xdg-open ../{{ streisand_local_directory }}/{{ streisand_server_name }}.html
+      ignore_errors: yes
+      when: hostvars['127.0.0.1']['ansible_system'] == "Linux"
+      become: no
+
+    - name: Open the instructions on OS X (if applicable)
+      local_action: command open ../{{ streisand_local_directory }}/{{ streisand_server_name }}.html
+      when: hostvars['127.0.0.1']['ansible_system'] == "Darwin"
+      become: no
+  when: not streisand_ci
```

Comment:
Is the reason that documentation can't be generated in CI that some of the roles have been removed and so the documentation fails on undefined variables? I believe that's likely the case but wanted to confirm. It would be nice to put a follow-up issue on the repo that would fix the documentation inter-dependencies since we will have to do that for the overall "modularity" work and it would also mean that CI could have docs generated. Testing docs in CI would be another great win!